hbase-server/src/main/java/org/apache/hadoop/hbase/util/TableDescriptorChecker.java

```diff
@@ -150,6 +151,12 @@ public static void sanityCheck(final Configuration c, final TableDescriptor td)
       warnOrThrowExceptionForFailure(logWarn, message, null);
     }
 
+    // Meta table shouldn't be set as read only, otherwise it will impact region assignments
+    if (td.isReadOnly() && TableName.isMetaTableName(td.getTableName())) {
+      String message = "Meta table can't be set as read only.";
+      warnOrThrowExceptionForFailure(false, message, null);
```

Comment:
Super nit/NOT IMPORTANT/NEXT TIME: just pass in the message rather than save to local variable.